include/onnc/ADT/NodeIterator.h

```diff
@@ -159,13 +159,15 @@ NodeIterator<NodeType, Traits>& NodeIterator<NodeType, Traits>::operator--()
 template<typename NodeType, template<typename N = NodeType> class Traits>
 NodeIterator<NodeType, Traits> NodeIterator<NodeType, Traits>::operator++(int)
 {
-  return NodeIterator(node()->getNextNode());
+  m_pNodePtr = m_pNodePtr->next;
+  return NodeIterator(node()->getPrevNode());
```

Comment:
The `NodeIterator` models [*LegacyBidirectionalIterator*](https://en.cppreference.com/w/cpp/named_req/BidirectionalIterator) concept, a pair of *LegacyBidirectionalIterator* can represent two kind of valid ranges:

1. **Go forward**: *[std::begin(c), std::end(c))* 
2. **Go backward**: *[std::prev(std::end(c)), std::prev(std::begin(c)))*

But you can't go back once reach one of the two positions (all the *NodeType* implementations won't let you do):

1. after last node
2. before first node

Thus you can't perform *operator--(int)* on the iterator points to **1** or *operator++(int)* on iterators points to **2**

A better way to implement postfix ++/-- is the employee copy constructor and prefix ++/-- together

```c++
NodeIterator copy(*this);
++(*this);
return copy; // possibly enable RVO here, no need std::move()
```